common/path_finder.h

```diff
@@ -110,6 +113,14 @@ class path_finder {
 
   inline path_finder &operator=(path_finder &&other);
 
+  /// Returns whether goto into the unkown is allowed.
+  bool are_unknown_tiles_allowed() const
```

Comment:
Just a small nitpick/comment/suggestion should this be called get_unknown_tiles_allowed() to be consistent with the getter-setter idiom in cpp ? I'm also ok if we consistently name all setters with a question like this. Then it needs to go into coding style and the lot, getter-setter is more standard.